### ISSUE TITLE
PDBList() download in a parallelized for loop results in skipping some files

### ISSUE THREAD

**Issue by uros-sipetic** [biopython/biopython#4092]
### Setup

I am reporting a problem with Biopython version  1.79, Python version 3.9, and operating
system Ubuntu 20.04:

I'm trying to download PDB files from a list of PDB IDs that I have, and doing so in a multi process for loop. However, after I start each new run, I get the message "Desired structure doesn't exists" randomly for some of the PDB IDs, and others finish successfully. I then rerun the script, and some of the ones that previously failed now successfully download, while others don't. And so I keep rerunning the script until I get all files. 

Is there an API rate limit that I'm hitting, hence the parallel process for loop is not allowed here...? A quick glance at the source code, and I see that the error message "Desired structure doesn't exists" is defaulted to for every download error (whether the structure exists or not, or if something else is the problem)

**Comment by sean-workman** [comment]
Could you post the code, or the relevant snippet, that you were running?

**Comment by uros-sipetic** [comment]
Yes of course, I apologize for not doing so in the original post

```
from Bio.PDB import PDBList
from joblib import Parallel, delayed

remote_server = 'ftp://ftp.wwpdb.org'
pdb_download_path = '/my/download/directory/'
pdb_ids = ['102L','102M','103L',103M','104L','104M','105M','106M','107L']  # this is a large list, containing thousands of PDB IDs

def process(pdb_id):
    pdbl = PDBList(remote_server)
    pdb_file_path = pdbl.retrieve_pdb_file(pdb_id, pdir=pdb_download_path, file_format='pdb')
Parallel(n_jobs=-1, verbose=1)(delayed(process)(pdb_id) for pdb_id in pdb_ids)
```

**Comment by uros-sipetic** [comment]
Hm Update:
Now even if downloading from https://files.rcsb.org/ I am having error - when trying locally on my computer, it works fine, but when I tried on a cluster to download 200 PDBs in parallel, every download failed with the error "Desired structure doesn't exists". 
Is there a rate limit to downloading here? Could it be that my cluster download was banned because of too many requests?
